compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -138,8 +138,12 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
     /** Is a reference to a class but not `this.type` */
     def isClassRef = sym.isClass && !tp.isInstanceOf[ThisType]
 
-    if (sym.exists && !sym.isStaticOwner && !isClassRef && !levelOK(sym))
+    if (!sym.exists || levelOK(sym))
+      None
+    else if (!sym.isStaticOwner && !isClassRef)
       tryHeal(sym, tp, pos)
+    else if (!sym.owner.isStaticOwner) // local class reference that is phase
+      levelError(sym, tp, pos, "")
```

Comment:
Will a member class be accepted or not? Maybe add a test for that.

Reply:
They will not. I added a regression test for it.

Reply:
Maybe update the comment too --- it's not only about local classes.

Reply:
Done